contracts/platform/include/platform/platform.hpp

```diff
@@ -59,6 +58,30 @@ using game_table = eosio::multi_index<
                    >;
 
 
+static uint64_t get_token_pk(const std::string& token_name) {
+    // https://github.com/EOSIO/eosio.cdt/blob/1ba675ef4fe6dedc9f57a9982d1227a098bcaba9/libraries/eosiolib/core/eosio/symbol.hpp
+    uint64_t value = 0;
+    for( auto itr = token_name.rbegin(); itr != token_name.rend(); ++itr ) {
+        if( *itr < 'A' || *itr > 'Z') {
+            eosio::check( false, "only uppercase letters allowed in symbol_code string" );
+        }
+        value <<= 8;
+        value |= *itr;
+    }
+    return value;
+}
+
+struct [[eosio::table("token"), eosio::contract("platform")]] token_row {
+    std::string token_name;
+    name contract;
+
```

Comment:
can we add precision field? asset = amount + symbol(symbol_code + precision)

Reply:
we could do it but there's a nice little function here
https://github.com/EOSIO/eosio.cdt/blob/master/libraries/eosiolib/core/eosio/asset.hpp#L360